Add docker rootless feature flag and its implementation for supporting docke rootless environment

kimsehwan96 opened the pull request:
resolve: #817 

Add feature flag `dockerRootless` and default value `false`. 

Change `lib/pip.js` file permission change logic for supporting docker rootless environment. 

I think that its better add some document about this in `README.md`

pgrzesik reviewed (commented):
Hey @kimsehwan96 👋 Thanks a lot for your contribution - do you think it would be possible to cover this by test by any chance, or it's gonna be rather hard to include in the testing suite?

kimsehwan96 commented:
Hello @pgrzesik . I think that this change should be tests in CI(Github action) level. Becasue it doesn't work that we expected when enable that feature flag in `Not` rootless docker environment. 

I will find how to test it in unit test or CI level test in github action. I think that it will not very hard to include this in the testing suite!

pgrzesik commented:
Hey @kimsehwan96 👋 Could you fix the formatting as listed in the failed CI build? I think having automated tests for this would be tricky, but let's make sure it's not breaking anything else in the process 👍

kimsehwan96 commented:
Hello @pgrzesik . Sorry for late works.. As you mentioned, apply automated tests for this chage is tricky. I underestimated about it 🥲. And I failed to devote much of time. 

By the way, I will applying the formatting and inform about it to you. 

+ When I have some free time from the coming week, I will append some comment about which cases of environment this will be needed with some detailed information. 

Thank you !

kimsehwan96 commented:
I applied formatting for this change ! @pgrzesik 

I want to merge 2 commits which are for applying `dockerRootless` option and `prettier` . So I rebased and squashed 2 commits into 1 and force pushed to this branch. Is it okay?

pgrzesik reviewed (approved):
Thank you @kimsehwan96 🙇
